src/inc/opcodes.h

```diff
@@ -233,12 +236,13 @@ STARTOP(put_subs_8bit_obs)
     const int           slot_ref     = int8(param[0]);
     const unsigned int  input_class  = uint8(param[1]),
                         output_class = uint8(param[2]);
-    uint16 index;
+    int index;
     slotref slot = slotat(slot_ref);
     if (slot)
     {
         index = seg.findClassIndex(input_class, slot->gid());
-        is->setGlyph(&seg, seg.getClassGlyph(output_class, index));
+        asset(index >= 0);
+        is->setGlyph(&seg, seg.getClassGlyph(output_class, static_cast<uint16>(index)));
```

Comment:
Please stop slamming asserts in everywhere where you have just switched from size_t to int because you know it may well return something < 0! If you static cast from int to uint16 what's so bad about having the cast occur earlier?
Graphite does not throw exceptions. That is a design constraint to allow it to have a pure C interface and not be dependent on libstdc++.